src/main/java/org/kohsuke/github/GitHub.java

```diff
@@ -276,56 +275,92 @@ public GHRateLimit getRateLimit() throws IOException {
     public GHMyself getMyself() throws IOException {
         requireCredential();
 
-        GHMyself u = retrieve().to("/user", GHMyself.class);
+        // This entire block is under synchronization to avoid the relatively common case
+        // where a bunch of threads try to enter this code simultaneously.  While we could
+        // scope the synchronization separately around the map retrieval and update (or use a concurrent hash)
+        // map, the point is to avoid making unnecessary GH API calls, which are expensive from
+        // an API rate standpoint
+        synchronized (users) {
+            GHUser cachedUser = users.get(this.login);
```

Comment:
@kohsuke This is invalid right now (we call getMyself in the constructor).  I'm fixing this to check for a valid login, but in the meantime don't merge this.